_guidelines-v4/04-cmn/03-cmnadditionalmeasurecontent/02-cmnslurties.md

```diff
@@ -65,17 +65,17 @@ Another reason for using elements instead of attributes for ties, slurs, and phr
 
 Although these elements are allowed within a {% include link elem="layer" %} to accommodate unmeasured notation, by convention in CMN they are normally placed inside {% include link elem="measure" %}, after the encoding of staves, alongside other so-called ‘control events’.
 
-{% include mei example="cmn/cmn-sample122.xml" valid="" %}
+{% include mei example="cmn/cmn-sample122.txt" valid="" %}
 
 Obviously, to be complete the slur in the above example needs to be ‘attached’ to the notes somehow. The ‘vertical assignment’ can be indicated for the example above using the **@staff** and **@layer** attributes like so:
 
-{% include mei example="cmn/cmn-sample123.xml" valid="" %}
+{% include mei example="cmn/cmn-sample123.txt" valid="" %}
 
 For the ‘horizontal assignment’, the encoder may choose between two different mechanisms. The first uses two timestamp attributes as described in section {% include link id="cmnTstamp" %}. The start and end points of the slur may be indicated thusly:
 
-{% include mei example="cmn/cmn-sample124.xml" valid="" %}
+{% include mei example="cmn/cmn-sample124.txt" valid="" %}
 
-By using **@tstamp** and **@tstamp2** attributes, the encoder denotes a rather loose connection – the slur (or tie) is attached to a certain position in the measure, not to a specific note or chord. If the encoder wants to specify a close connection to a particular event, the **@startid** and **@endid** attributes may be used instead. Here, the **@xml:id**s of the first and last note of the slur are referenced. This mechanism also allows the crossing of layers and staves. 
+By using **@tstamp** and **@tstamp2** attributes, the encoder denotes a rather loose connection – the slur (or tie) is attached to a certain position in the measure, not to a specific note or chord. If the encoder wants to specify a close connection to a particular event, the **@startid** and **@endid** attributes may be used instead. Here, the **@xml:id**s of the first and last note of the slur are referenced. This mechanism also allows the crossing of layers and staves.
 
 For human readability, it is recommended to encode {% include link elem="slur" %}, {% include link elem="tie" %} and {% include link elem="phrase" %} features in the {% include link elem="measure" %} where they begin; that is, in the measure that holds the element referenced by **@startid**. On the other hand, for machine processability, it may be desirable to place {% include link elem="slur" %}, {% include link elem="tie" %}, and {% include link elem="phrase" %} elements in the measure *where they end* or even in the *last measure* regardless of their beginning and ending points in the music. This last option makes all references contained within these elements ‘back references’. Back references are necessary when using processing software that treats the encoded file as a stream; that is, programs that process the file without creating an in-memory representation of its contents.
```

Comment:
should we really include information about processing models? I'm imclined to completely drop stuff like that…

Reply:
Yes. Also because placing control events not in the measure where they start rules out the use of `@tstamp` and `@tstamp2`.